### PR TITLE
Fix races in LifetimeDefinition

### PR DESCRIPTION
**Pull request by ulex**
This pull request fixes two issues regarding CancellationTokenSource in LifetimeDefinition:

  * First issue with destruction may lead to the null reference exception when cancellation token is being requested on terminating lifetime (kudos to @Denis-Korneev for noticing the failed test on teamcity and suggested fix)
 * Second issue is that not all subscribers are guaranteed to be notified about cancellation because of the error in the initialization of myCts.

### CODE REVIEW

**Review by ForNeVeR — APPROVED** [review]
LGTM, but waiting for thorough review from @Iliya-usov.

**Review by Iliya-usov — APPROVED** [review]

**Pull request merged**
